contrib/endpoints/src/api_manager/method_impl.h

```diff
@@ -175,6 +183,9 @@ class MethodInfoImpl : public MethodInfo {
 
   // Whether the response is streaming or not.
   bool response_streaming_;
+
+  // map of metric and its cost
+  std::map<std::string, int64_t> metric_cost_map_;
```

Comment:
std::map using balanced tree to maintain order and lookup. if we don't need an item order, unordered_map is more efficient. and if we don't lookup, vector is better for iteration.